src/components/Map/LeafletMap.jsx

```diff
@@ -227,6 +215,113 @@ MapView.propTypes = {
   coordinates: PropTypes.arrayOf(PropTypes.number),
 };
 
+/**
+ * Custom Leaflet location control, for "use current user location" functionality.
+ *
+ * @class
+ * @constructor
+ */
+const LocationButton = Leaflet.Control.extend({
+  /**
+   * Create a location button.
+   *
+   * @param disabledButton boolean
+   * @param onClick fn - a function that doesn't accept arguments
+   * @param tooltip string
+   * @param ariaLabel string
+   */
+  initialize: function (disabledButton, onClick, tooltip, ariaLabel) {
+    this.disabledButton = disabledButton;
+    this.onClick = onClick;
+    this.tooltip = tooltip;
+    this.ariaLabel = ariaLabel;
+  },
+  /**
+   * Global configuration for the control.
+   */
+  options: {
+    position: 'topleft',
+  },
+  /**
+   * Define the html and functionality of the control button.
+   *
+   * @returns {HTMLDivElement}
+   */
+  onAdd: function () {
+    const container = Leaflet.DomUtil.create('div', 'leaflet-bar leaflet-control');
+    const button = Leaflet.DomUtil.create(
+      'a',
+      `leaflet-control-button fa-solid fa-location-crosshairs ${this.disabledButton && 'leaflet-disabled'}`,
+      container
+    );
+
+    Leaflet.DomEvent.disableClickPropagation(button);
+    Leaflet.DomEvent.on(button, 'click', () => {
+      if (!this.disabledButton) {
+        this.onClick();
+      }
+    });
+
+    button.title = this.tooltip;
+    button.setAttribute('role', 'button');
+    button.setAttribute('aria-disabled', this.disabledButton);
+    button.setAttribute('aria-label', this.ariaLabel);
+
+    return container;
+  },
+});
+
+const LocationControl = () => {
```

Comment:
https://github.com/open-formulieren/open-forms-sdk/pull/821/files#r2047173243 this is also a prime candidate for the `createControlComponent` helper I think!